workshops/painting_app/painting_app.md

```diff
@@ -0,0 +1,323 @@
+- This function takes the context of the canvas element and using the property `lineWidth`, it sets the thickness of the brush to the value from the range input.
+
+Then, we'll need to handle the colors of the brush, in this block of code we'll make sure that whenever the user clicks on any of the color buttons, the color of the brush changes.
+```js
+// Handle Colors
+var colors = document.getElementsByClassName('colors')[0];
+
+colors.addEventListener('click', function(event) {
+ context.strokeStyle = event.target.value || 'black';
+});
+```
+- First, we are creating a variable that uses the method `getElementByClassName` to get an element from the document with the specified class. It gets the `div` element from the document, this div container acts like an array and every element inside of it is indexed.
+- Then, we are taking the `colors` variable (div element), and we are adding it an `addEventListener`, every time a button gets clicked a function runs.
+- Ultimately, this function takes the context of the canvas element and using the property `strokeStyle`, it sets the color of the brush to one of the button's value or to black.
+
+Now, we are going to create 3 events that depending on if the mouse is on the canvas, will draw.
+```js
+	// Mouse Down Event
+	canvas.addEventListener('mousedown', function(event) {
+		setMouseCoordinates(event);
+		isDrawing = true;
+
+		// Start Drawing
+		context.beginPath();
+		context.moveTo(mouseX, mouseY);
+	});
+
+	// Mouse Move Event
+	canvas.addEventListener('mousemove', function(event) {
+		setMouseCoordinates(event);
+
+		if(isDrawing){
+		  context.lineTo(mouseX, mouseY);
+		  context.stroke();
+		}
+	});
+
+	  // Mouse Up Event
+	canvas.addEventListener('mouseup', function(event) {
+		setMouseCoordinates(event);
+		isDrawing = false;
+  });
+```
+
+The first event, takes the canvas element and adds it an event listener, when the event has place, the function starts running. What this function does is to call the function `setMouseCoordinates` (which i won't explain right now but don't worry we'll see it later), and then it asigns the value of `isDrawing` to `true`, finally the function takes the context of the canvas element, and uses the `beginPath` method to start a new path, and then it also takes the `moveTo` method to start a sub-path in the specified coordinates.
+
+We do almost the same thing for the mouse move event; in this one we don't have to re-assign the value of the variable `isDrawing` though, because we already did, and then we use an if statement to run the code between the curly braces if the `isDrawing` variable has a true value (which is the case); what this code does, is to take the context of the canvas element and using the methods `lineTo` and `stroke`, we add a straight line to the current sub-path by connecting the sub-path's last point to the specified (x, y) coordinates, and we stroke the current or given path with the current stroke style.
```

Comment:
```suggestion
We do almost the same thing for the mouse move event; but here, we don't have to re-assign the value of the variable `isDrawing` though, because we already did, and then we use an if statement to check if the `isDrawing` variable has a true value (which is the case) or not. If it is true, it takes the context of the canvas element and using the methods `lineTo` and `stroke`, we add a straight line to the current sub-path by connecting the sub-path's last point to the specified (x, y) coordinates, then we stroke the current or given path with the current stroke style.
```